sympy/polys/matrices/domainmatrix.py

```diff
@@ -213,6 +212,55 @@ def from_list_sympy(cls, nrows, ncols, rows, **kwargs):
 
         return DomainMatrix(domain_rows, (nrows, ncols), domain)
 
+    @classmethod
+    def from_dict_sympy(cls, nrows, ncols, elemsdict, **kwargs):
+        """
+
+        Parameters
+        ==========
+
+        nrows: number of rows
+        ncols: number of cols
+        elemsdict: dict of dicts containing non-zero elements of the DomainMatrix
+
+        Returns
+        =======
+
+        DomainMatrix containing elements of elemsdict
+
+        Examples
+        ========
+
+        >>> from sympy.polys.matrices import DomainMatrix
+        >>> from sympy import QQ, ZZ
+        >>> elemsdict = {0: {1: QQ(1, 2)}, 1: {0: ZZ(1)}}
+        >>> DomainMatrix.from_dict_sympy(2, 3, elemsdict)
```

Comment:
This is supposed to be from SymPy types not from domain elements

Reply:
Ok I'll change the example and use Sympy types instead of domain elements